src/main/java/org/aludratest/service/jms/data/JmsMessageData.java

```diff
@@ -23,5 +26,35 @@
  */
 
 public abstract class JmsMessageData extends Data {
-	// empty parent interface
+	// parent class
+	
+	/**
+	 * Properties to apply the jms message.
+	 * Supports the properties of javax.jms.Message
+	 */
+	private Map<String, Object> properties;
+	
+	JmsMessageData() {
+		//default, empty property list
+		properties = new HashMap<String, Object>();
+	}
+
+	public Map<String, Object> getProperties() {
+		return properties;
+	}
```

Comment:
Allows direct, uncontrolled access and modification to the internal map. Should only offer methods for getting and setting a single property. OR return a copy of the map, and in the setter, fill the map with the contents of the passed map.

Reply:
Hmm... `return Collections.unmodifiableMap(properties)` would have done it :-)